data/scrape/lib/generate_summary.py

```diff
@@ -114,10 +112,69 @@ def main() -> None:
                 f"| {count:,} |",
                 file=sink,
             )
+    # adding KPI code
+    # Currently hard-coded; what's a good alternative?
+    TSV_PATH = "../tsv_summary.tsv"
+    table = pandas.read_csv(
+        TSV_PATH,
+        sep="\t",
+        names=[
+            "link",
+            "iso_639_2_code",
+            "iso_639_language_name",
+            "wiktionary_language_name",
+            "script",
+            "dialect",
+            "filtered",
+            "narrow_broad",
+            "case_folding",
+            "no_entries",
+        ],
+    )
+    # Somewhat arbitrarily, we use the ISO 639-2 code as the "real" label for
+    # languages.
+    languages = table.iso_639_2_code.nunique()
+    print(f"Languages:\t{languages:,}")
+    broad_files = table["narrow_broad"].value_counts()["Broad"]
+    narrow_files = table["narrow_broad"].value_counts()["Narrow"]
+    # We don't currently have any case where multiple languages have a
+    # "southern" vs. a "northern" dialect, but just in case, we use this
+    # slightly more complex logic here.
+    dialects = table.groupby(["iso_639_2_code", "dialect"]).ngroups
+    print(f"Dialects:\t{dialects:,}")
+    df = table[table["dialect"] != ""]
+    grouped_counts = (
+        df.groupby(["iso_639_2_code", "dialect", "narrow_broad"])
+        .size()
+        .unstack(fill_value=0)
+    )
+    narrow_pairs_files = grouped_counts["Narrow"].sum()
+    broad_pairs_files = grouped_counts["Broad"].sum()
+    scripts = table.script.nunique()
+    print(f"Scripts:\t{scripts:,}")
+    # We only count "unfiltered" pronunciations, since filtered ones are a
+    # strict subset thereof. Note that this gives us a smaller number than
+    # previously reported, but it seems more correct.
+    entries = table[~table.filtered].no_entries.sum()
+    print(f"Entries:\t{entries:,}")
+
+    bullet_list = (  
+        f"* Languages: {languages} \n"
+        f"\t* Broad transcription files: {broad_files} \n"
+        f"\t* Narrow transcription files: {narrow_files} \n"
+        f"* Languages & dialects: {dialects} \n"
+        f"\t* Broad transcription files: {broad_pairs_files} \n"
+        f"\t* Narrow transcription files: {narrow_pairs_files} \n"
+        f"* Scripts: {scripts} \n"
+        f"* Pronunciations: {entries:,}\n\n"
+    )
+    with open(README_PATH, "r") as original:
```

Comment:
Generally speaking this is not a great design---it's sort of hackish to write the Markdown and TSV files, read the TSV file again, compute some summary statistics, and prepend them to the Markdown file...

What I suggest is this:

1. Write the TSV file; compute the KPIs as you go (this might get rid of the Pandas dependency too).
2. Write the KPIs to the Markdown file.
3. Write the remaining information to the Markdown file (you can in that case read the data directly from the TSV file if you want).